python/lsst/afw/display/utils.py

```diff
@@ -370,3 +370,31 @@ def drawCoaddInputs(exposure, frame=None, ctype=None, bin=1, display="deferToFra
                             for point in ccdCorners]
             display.line([(coaddCorners[i].getX()/bin, coaddCorners[i].getY()/bin)
                       for i in range(-1, 4)], ctype=ctype)
+
+def drawSourceMatches(exposure, sources, matches, frame):
+    """Display an Exposure with overplotted sources and reference catalog matches.
+
+    Source positions are indicated by green circles; catalog positions by
+    yellow vertical crosses (+); match positions by red diagonal crosses (X).
+
+    @param[in] exposure  Exposure to display (lsst.afw.image.Exposure)
+    @param[in] sources   Sources to overplot (lsst.afw.table.SourceCatalog)
+    @param[in] matches   Matches to overplot (lsst.afw.table.ReferenceMatch)
+    @param[in] frame     Frame to use for display
+    """
+    disp = _getDisplayFromDisplayOrFrame(None, frame)
+    disp.mtv(exposure)
+    x0, y0 = exposure.getMaskedImage().getX0(), exposure.getMaskedImage().getY0()
+    wcs = exposure.getWcs()
+
+    with disp.Buffering():
+        for source in sources:
+            x, y = source.getX() - x0, source.getY() - y0
+            disp.dot('o', x, y, ctype="green", size=4)
```

Comment:
You shouldn't need to deal with XY0 yourself (it's handled by the origin=XXX keyword).   Did you check that you needed to?